src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_errors.c

```diff
@@ -46,10 +41,12 @@ xocl_insert_error_record(struct xocl_dev_core *core, struct xclErrorLast *err_la
 int
 xocl_init_errors(struct xocl_dev_core *core)
 {
-	/*TODO
-	 * mutex_init(core->errors_lock);
-	 */
-	core->errors = NULL;
+	mutex_init(&core->errors_lock);
```

Comment:
should be mutex_lock(). And actually, please remove lock/unlock calls from _probe/_remove code, because all the extra thread should be exit at this time.

Reply:
done